python/ray/experimental/data/dataset.py

```diff
@@ -346,59 +346,182 @@ def input_files(self) -> List[str]:
         """
         raise NotImplementedError  # P0
 
-    def write_parquet(path: str,
-                      filesystem: Optional["pyarrow.fs.FileSystem"] = None
-                      ) -> None:
+    def write_parquet(
+            self,
+            paths: Union[str, List[str]],
+            filesystem: Optional["pyarrow.fs.FileSystem"] = None) -> None:
         """Write the dataset to parquet.
 
         This is only supported for datasets convertible to Arrow records.
 
         Examples:
+            # Write to a single file.
             >>> ds.write_parquet("s3://bucket/path")
 
+            # Write to multiple files.
+            >>> ds.write_parquet(["/path/to/file1", "/path/to/file2"])
```

Comment:
We shouldn't support this--- output should always be a directory in which we write each block as a separate file.